### PR TITLE
Improve godoc for MultiClusterServiceDNSRecord and DNSEndpoint

### PR DESCRIPTION
**Pull request by pmorie**
Adds basic information about service DNS to the API godoc. Since ingress is very similar, I'll make this for DNS only for now, get the verbiage correct in this PR, and then do a follow-up PR to add godoc for ingress as well.

### CODE REVIEW

**Comment by pmorie** [comment]
I'll follow this PR up with similar doc for ingress

File: pkg/apis/multiclusterdns/v1alpha1/multiclusterservicednsrecord_types.go [hunk 1]
```diff
@@ -51,7 +52,31 @@ type ClusterDNS struct {
...
+//
+// For the example service mentioned earlier this would yield:
+//
+// test-service.test-namespace.(spec.federationName).svc
+//
+// Additionally, subdomains of this DNS name will be programmed for endpoints on
+// a region and zone basis:
+//
+// (status.DNS[*].region).test-service.test-namespace.(spec.federationName).svc
+// (status.DNS[*].zone).(status.DNS[*].region).test-service.test-namespace.(spec.federationName).svc
```
Comment (shashidharatd) [inline pkg/apis/multiclusterdns/v1alpha1/multiclusterservicednsrecord_types.go]: Thanks @pmorie for this godoc.
This is slightly incorrect. There will be 3 names per cluster service instance.
and naming scheme is as follows:
1. Zone Level - (metadata.name).(metadata.namespace).(spec.federationName).svc.(status.DNS[\*].zone).(status.DNS[*].region).(spec.DNSSuffix)
2. Region Level - (metadata.name).(metadata.namespace).(spec.federationName).svc.(status.DNS[*].region).(spec.DNSSuffix)
3. Global Level - (metadata.name).(metadata.namespace).(spec.federationName).svc.(spec.DNSSuffix)

**Comment by pmorie** [comment]
@danehans 

> Is there a reason why this PR does not cover `multiclusteringressdnsrecord_types.go`?

Since they're very similar, I wanted to get the verbiage correct here and then make another PR for ingress.

**Comment by danehans** [comment]
@pmorie I created Issue #317 to track the follow-on effort for godoc'ing `MultiClusterIngressDNSRecord`

File: pkg/apis/multiclusterdns/v1alpha1/dnsendpoint_types.go [hunk 2]
```diff
@@ -30,17 +30,17 @@ type TTL int64
 // it is then stored in a persistent storage via serialization
 type Labels map[string]string
 
-// Endpoint is a high-level way of a connection between a service and an IP
+// Endpoint is a high-level connection between a service and an IP.
```
Comment (danehans) [inline pkg/apis/multiclusterdns/v1alpha1/dnsendpoint_types.go]: I would consider changing `connection` to `association` or `mapping`. When I see connection, I think of network plumbing.

File: pkg/apis/multiclusterdns/v1alpha1/dnsendpoint_types.go [hunk 3]
```diff
@@ -30,17 +30,17 @@ type TTL int64
...
 type Endpoint struct {
-	// The hostname of the DNS record
+	// The hostname of the DNS record.
```
Comment (danehans) [inline pkg/apis/multiclusterdns/v1alpha1/dnsendpoint_types.go]: I thought this is the FQDN of the DNS resource record?

File: pkg/apis/multiclusterdns/v1alpha1/dnsendpoint_types.go (hunk 2 above)
Comment (pmorie) [reply pkg/apis/multiclusterdns/v1alpha1/dnsendpoint_types.go]: Good point

**Comment by pmorie** [comment]
@danehans incorporated your feedbacks and rebased

cc @shashidharatd

**Comment by shashidharatd** [comment]
/lgtm
Thanks @pmorie !

**Pull request merged**